ra2ce/analysis/losses/losses_base.py

```diff
@@ -295,52 +302,49 @@ def _create_result(vlh: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
-                    vlh_row["u"],
-                    vlh_row["v"],
-                    key,
-                )
-                # allow link_id not to be unique in the graph (results reliability is up to the user)
-                # this can happen for instance when a directed graph should be made from an input network
-                for performance_row in row_performance_changes.iterrows():
-                    row_performance_change = performance_row[-1][
-                        f"{self.performance_metric}"
+                if vlh_row[event] > self.analysis.threshold:
+                    row_hazard_range = _get_range(vlh_row[event])
+                    row_connectivity = vlh_row[connectivity_attribute]
+                    row_performance_changes = performance_change.loc[
+                        [vlh_row[self.link_id]]
                     ]
-                    if "key" in performance_row[-1].index:
-                        performance_key = performance_row[-1]["key"]
+                    if "key" in vlh_row.index:
+                        key = vlh_row["key"]
                     else:
-                        performance_key = 0
-                    row_u_v_k = (
-                        performance_row[-1]["u"],
-                        performance_row[-1]["v"],
-                        performance_key,
-                    )
-                    if (
-                        math.isnan(row_performance_change) and row_connectivity == 0
-                    ) or row_performance_change == 0:
-                        self._calculate_production_loss_per_capita(
-                            vehicle_loss_hours, row_hazard_range, vlh_row, event
-                        )
-                    elif not (
-                        math.isnan(row_performance_change)
-                        and math.isnan(row_connectivity)
-                    ) and ((u, v, k) == row_u_v_k):
-                        self._populate_vehicle_loss_hour(
-                            vehicle_loss_hours,
-                            row_hazard_range,
-                            vlh_row,
-                            row_performance_change,
-                            event,
+                        key = 0
+                    (u, v) = (vlh_row["u"], vlh_row["v"])
+                    # allow link_id not to be unique in the graph (results reliability is up to the user)
+                    # this can happen for instance when a directed graph should be made from an input network
+                    for performance_row in row_performance_changes.iterrows():
+                        row_performance_change = performance_row[-1][
+                            f"{self.performance_metric}"
+                        ]
+                        if "key" in performance_row[-1].index:
+                            performance_key = performance_row[-1]["key"]
+                        else:
+                            performance_key = 0
+                        row_u_v_k = (
```

Comment:
You are only using this once, memory wise is not the best approach, you could just simply refer to the concrete locations in line 340